### PR TITLE
[CoroutineAccessors] Add read and modify.

### PR DESCRIPTION
**Pull request by nate-chandler**
Enable parsing the new keywords behind the CoroutineAccessors experimental feature.

### CODE REVIEW

**Comment by nate-chandler** [comment]
https://github.com/swiftlang/swift/pull/76526

@swift-ci please test

**Comment by nate-chandler** [comment]
https://github.com/swiftlang/swift/pull/76526

@swift-ci please test

File: CodeGeneration/Sources/SyntaxSupport/KeywordSpec.swift [hunk 1]
```diff
@@ -628,6 +632,8 @@ public enum Keyword: CaseIterable {
       return KeywordSpec("protocol", isLexerClassified: true)
     case .public:
       return KeywordSpec("public", isLexerClassified: true)
+    case .read:
+      return KeywordSpec("read")
```
Comment (ahoppen) [inline CodeGeneration/Sources/SyntaxSupport/KeywordSpec.swift:636]: Should `read` also be behind an experimental feature, like `modify`?

File: Tests/SwiftParserTest/DeclarationTests.swift [hunk 2]
```diff
@@ -3334,4 +3334,20 @@ final class DeclarationTests: ParserTestCase {
...
+          yield _i
+        }
+        modify {
+          yield &_i
+        }
+      }
+      """,
+      experimentalFeatures: .coroutineAccessors
+    )
+  }
```
Comment (ahoppen) [inline Tests/SwiftParserTest/DeclarationTests.swift:3352]: Could you also add a test case that `read` and `modify` are not parsed when the `.coroutineAccessors` experimental features is not specified?

File: CodeGeneration/Sources/SyntaxSupport/KeywordSpec.swift (hunk 1 above)
Comment (nate-chandler) [reply CodeGeneration/Sources/SyntaxSupport/KeywordSpec.swift:636]: Thanks!  Fixed.

File: Tests/SwiftParserTest/DeclarationTests.swift (hunk 2 above)
Comment (nate-chandler) [reply Tests/SwiftParserTest/DeclarationTests.swift:3352]: Yep--added one for each.

**Comment by nate-chandler** [comment]
Accidentally opened this PR from a branch in this repo rather than my fork.  Closing in favor of the one from my fork: https://github.com/swiftlang/swift-syntax/pull/2855 .

**Pull request closed without merging**